recipes-extended/i2c-quark-board/files/i2c-quark-apds9960.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * APDS9960 I2C sensor test file
+ * build command: KDIR=/kernel-source/ make
+ * The kernel config should include CONFIG_APDS9960=m
+ */
+
+#include <linux/module.h>
+#include <linux/gpio.h>
+#include <linux/i2c.h>
+
+#define APDS9960_I2C_ADDR		0x39
+
+/* The IRQ pin is connected to GPIO14=IO3
+ */
+#define APDS9960_I2C_IRQ		14
+
+static struct i2c_board_info i2c_device = {
+	I2C_BOARD_INFO("apds9960", APDS9960_I2C_ADDR),
+	.irq = -1,
+};
+
+static struct i2c_client *i2c_client;
+static int __init apds9960_init(void)
+{
+	int i = 0;
+	struct i2c_adapter *adap = NULL;
+
+	for (i = 0; i < 10; i++) {
```

Comment:
I just want to enumerate all i2c adapters and find the one connected to external devices,  but I did not find the API for the total i2c adapters. Since the total number is board related, and Galileo boards only have one i2c adapter, I changed the code to use adapter 0.